proposals/p1983.md

```diff
@@ -0,0 +1,183 @@
+
+Note that any regular grouping rule can present similar issues for Indian digit
+grouping conventions.
+
+[Proposal #143: Numeric literals](/proposals/p0143.md) chose 3-digit decimal
+groupings.
+
+Given there are overall advantages to not enforcing regular digit conventions,
+including for hex digits, it seems unnecessary to conform to the currently
+established decimal conventions. While this removes the enforcement, the
+resulting accidents are considered a reasonable risk.
+
+In theory a code linter could be told to prefer certain formats with options to
+change behavior, although that may remain too low benefit to implement.
+
+### 2-digit or 4-digit hexadecimal digit groupings
+
+Hexadecimal digit groupings could be enforced along two axes:
+
+-   Every 2 (`F_FF`) or 4 (`F_FFFF`) characters, corresponding to one or two
+    bytes respectively.
+-   Regular or irregular placement. For example, if a digit separator is placed
+    every byte (every other digit), we could require regular placement every
+    byte as in `FF_FF_FF_FF`, or irregular as in `FF_FFFFF_FF` which skips one
+    placement.
+
+[Proposal #143: Numeric literals](/proposals/p0143.md) chose 4-digit with
+regular placement.
+
+Advantages:
+
+-   More enforcement of digit groupings can prevent bugs and misreadings
+    resulting from accidentally incorrect groupings.
+
+Disadvantages:
+
+-   Again, microformats become an issue.
+    -   As noted in background, MAC addresses are typically in byte pairs
+        (`00_B0_D0_63_C2_26`) and UUIDs have particular groupings
+        (`123E4567_E89B_12D3_A456_426614174000`).
+-   If all of the other literal formats (decimal, real, and binary) allow
+    arbitrary digit separator placement, enforcing placement in hexadecimal
+    numbers would be an outlier.
+
+While it may be that enforcing 2 character (one byte) groupings with irregular
+placement would prevent sufficient errors versus the developer inconvenience
+risks, it risks becoming an outlier and for only very loose enforcement.
+
+Similar to the preceding conclusion, leaving these issues to code linters may be
+best.
+
+### Disallow digit separators in fractions
+
+[Proposal #143: Numeric literals](/proposals/p0143.md) appears to disallow digit
+separators in fractions. That is, in `1.2345`, `1.23_45` is disallowed. This
+proposal changes that.
+
+Advantages:
+
+-   Reduces the chance of confusion resulting from mistaking a `.` for another
+    `_` separator, as in `1_234_567.890_123_456`.
+
+Disadvantages:
+
+-   While sometimes commas aren't written for digit separators, it's notable
+    that SI advice is to use digit spacing instead
+    ([Digit spacing, rule #16](https://physics.nist.gov/cuu/Units/checklist.html)).
+    As a consequence, it's likely common to want some kind of digit separator
+    after the decimal point.
+-   Would create an inconsistency in placement rules.
+
+We won't disallow digit separators in fractions.
```

Comment:
Avoid the double negative and just say that we allow them?

Reply:
Done